workers/unity/Assets/Playground/Scripts/MonoBehaviours/SpawnCubeCommandReceiver.cs

```diff
@@ -0,0 +1,112 @@
+{
+    public class SpawnCubeCommandReceiver : MonoBehaviour
+    {
+        [Require] private Transform.Requirables.Reader transformReader;
+        [Require] private CubeSpawner.Requirables.CommandRequestHandler commandRequestHandler;
+        [Require] private CubeSpawner.Requirables.Writer writer;
+        [Require] private WorldCommands.Requirables.WorldCommandRequestSender worldCommandRequestSender;
+        [Require] private WorldCommands.Requirables.WorldCommandResponseHandler worldCommandResponseHandler;
+
+        private ILogDispatcher logDispatcher;
+
+        private const string EntityForIdFailedWithMessage = "Create entity (for id {0}) failed with message: \"{1}\"";
+        private const string FailedToReserveEntityIdMessage = "Failed to reserve entity id: {0}";
+
+        private readonly HashSet<long> sentRequestIds = new HashSet<long>();
+
+        public void OnEnable()
+        {
+            commandRequestHandler.OnSpawnCubeRequest += OnSpawnCubeRequest;
+
+            worldCommandResponseHandler.OnReserveEntityIdsResponse += OnEntityIdsReserved;
+            worldCommandResponseHandler.OnCreateEntityResponse += OnEntityCreated;
+
+            logDispatcher = GetComponent<SpatialOSComponent>().LogDispatcher;
+        }
+
+        private void OnSpawnCubeRequest(CubeSpawner.SpawnCube.RequestResponder requestResponder)
+        {
+            requestResponder.SendResponse(new Empty());
+
+            sentRequestIds.Add(worldCommandRequestSender.ReserveEntityIds(1));
+        }
+
+        private void OnEntityIdsReserved(WorldCommands.ReserveEntityIds.ReceivedResponse response)
+        {
+            if (!sentRequestIds.Remove(response.RequestId))
+            {
+                // This response was not for a command from this behaviour.
+                return;
+            }
+
+            var responseOp = response.Op;
+
+            if (responseOp.StatusCode != StatusCode.Success)
+            {
+                logDispatcher.HandleLog(LogType.Error,
+                    new LogEvent(string.Format(FailedToReserveEntityIdMessage, responseOp.Message)));
+
+                return;
+            }
+
+            var location = transformReader.Data.Location;
+            var cubeEntityTemplate =
+                CubeTemplate.CreateCubeEntityTemplate(new Coordinates(location.X, location.Y + 2, location.Z));
+            var expectedEntityId = responseOp.FirstEntityId.Value;
+
+            sentRequestIds.Add(worldCommandRequestSender.CreateEntity(cubeEntityTemplate, expectedEntityId));
+        }
+
+        private void OnEntityCreated(WorldCommands.CreateEntity.ReceivedResponse response)
+        {
+            if (!sentRequestIds.Remove(response.RequestId))
+            {
+                // This response was not for a command from this behaviour.
+                return;
+            }
+
+            var createEntityResponseOp = response.Op;
+
+            if (createEntityResponseOp.StatusCode != StatusCode.Success)
```

Comment:
Nitpick - we aren't dealing with the case where the entity was created but the response timed out. I don't think we have to deal with it but worth noting there is an unhandled failure case here.
Another failure case is you losing authority before the response comes back. Again I don't think we have to deal with it.

Reply:
We won't have this playground by then :P

Reply:
Also we know how to solve it. The question is how correct or readable we want this code to be.